core/src/trezor/ui/layouts/common.py

```diff
@@ -24,11 +33,12 @@ async def button_request(
 
 
 async def interact(
-    ctx: wire.GenericContext,
     layout: LayoutType,
     br_type: str,
     br_code: ButtonRequestType = ButtonRequestType.Other,
+    ctx: wire.GenericContext | None = None,
```

Comment:
This is due to device recovery homescreen which is not a workflow so it doesn't go through `with_context`.

Reply:
Should be dealt with in c61ca19ffcdc0231ccc846ba544b3f4053f2ffbb & 2ae20f7e44eecd0f3811417fe7d4f7f3de454d88.

Without 2ae20f7e44eecd0f3811417fe7d4f7f3de454d88 `test_wrong_nth_word` test gets stuck.

Reply:
`test_same_share` does the same thing
and for the record it's not that the test itself is stuck -- instead, whatever testcase runs _after_ either test gets stuck and does not accept the first DebugLinkDecision

also if you interrupt the test suite in this state, the next `Initialize` message fails with `Cancelled`.

it's weird and something we should solve

Reply:
n.b., unlike other kinds of debuglink breakages, manually clicking through the one missing debuglinkdecision will fully resume the test suite

Reply:
i can reproduce the same stickiness when I manually set the Trezor emulator to recovery homescreen via `trezorctl device recover` and entering number of words, then interrupting recovery.

the same stickiness also happens if I shut down and then restore the emulator in recovery homescreen.

That means that while the recovery screen is running, as started via `with_context`, it will consume? the first DebugLinkDecision sent by the host. I recall seeing this exact problem before, and it is something that we should in theory be able to solve cleanly now.

Reply:
my current idea is, `CURRENT_LAYOUT` should be already set by the time the ButtonRequest is sent.